barretenberg/cpp/pil/vm2/bytecode/bc_hashing.pil

```diff
@@ -66,68 +67,168 @@ namespace bc_hashing;
     // to be activated which is impossible on first row.), LATCH_CONDITION is a boolean.
     pol LATCH_CONDITION = latch + precomputed.first_row;
 
-    // The start of a new bytecode id and new set of hashing runs. Needs to be a committed column as it is used in the lookup
-    pol commit start;
-    start * (1 - start) = 0;
+    // The start of a new bytecode id and new set of hashing runs:
+    pol START = sel * (1 - sel_not_start);
+
+    // Needs to be a committed column as it is used in the lookup:
+    pol commit sel_not_start;
+    sel_not_start * (1 - sel_not_start) = 0;
 
     // If the current row is a latch or the first row, the next row should be a start (if it's active)
     #[START_AFTER_LATCH]
-    sel' * (start' - LATCH_CONDITION) = 0;
+    sel' * ((1 - sel_not_start') - LATCH_CONDITION) = 0;
 
     // Used as part of the lookup into bytecode decomposition
-    pol commit pc_index;
-    // The PC increments by 31 each row as long as the row is not latched, in which case the next pc is zero
+    pol commit pc_index, pc_index_1, pc_index_2;
+    // The PC increments by 31*3=93 each row as long as:
+    // a) the row is not latched, in which case the next pc is zero
+    // b) we are not at start, in which case we only read 2 fields, and => only increment pc by 31*2=62
     #[PC_INCREMENTS]
-    sel * (pc_index' - ((1 - LATCH_CONDITION) * (31 + pc_index))) = 0;
+    (sel + precomputed.first_row) * (pc_index' - ((1 - LATCH_CONDITION) * (62 + pc_index + sel_not_start * 31))) = 0;
+    // At start, we need packed_fields[0] = sep:
+    #[PC_INCREMENTS_1]
+    sel * ( pc_index_1 - START * pc_index - sel_not_start * ( pc_index + 31 )) = 0;
+    #[PC_INCREMENTS_2]
+    sel * ( pc_index_2 - (pc_index_1 + 31) ) = 0;
 
     pol commit bytecode_id;
     #[ID_CONSISTENCY]
     (1 - LATCH_CONDITION) * (bytecode_id' - bytecode_id) = 0;
 
-    pol commit packed_field;
-    #[GET_PACKED_FIELD]
-    sel { pc_index, bytecode_id, packed_field }
+    // We need 3 packed fields to use as inputs for each poseidon round
+    pol commit packed_fields_0, packed_fields_1, packed_fields_2;
+
+    // At the start of a new bytecode hash, the initial field has to be the separator, and we skip the lookup:
+    #[START_IS_SEPARATOR]
+    START * (packed_fields_0 - constants.GENERATOR_INDEX__PUBLIC_BYTECODE) = 0;
+
+    #[GET_PACKED_FIELD_0]
```

Comment:
I think it's ok to have 3 lookups, at least until the world proves otherwise. `bc_decomposition` is BIG, but `sel_packed` is 31 times smaller. I have some faith.